test/src/expectEvent.truffle.test.js

```diff
@@ -27,18 +29,21 @@ contract('expectEvent (truffle contracts)', function ([deployer]) {
     context('short uint value', function () {
       it('accepts emitted events with correct BN', async function () {
         await expectEvent.inConstruction(this.emitter, 'ShortUint',
-          { value: new BN(this.constructionValues.uint) }
+          { value: new BN(this.constructionValues.uint) },
         );
       });
 
       it('throws if a correct JavaScript number is passed', async function () {
-        await assertFailure(
-          expectEvent.inConstruction(this.emitter, 'ShortUint', { value: this.constructionValues.uint })
+        const { message } = await assertFailure(
+          expectEvent.inConstruction(this.emitter, 'ShortUint', { value: this.constructionValues.uint }),
         );
+        expect(message).to.match(errorRegex);
```

Comment:
The error here isn't that the values don't match, but that the _types_ are wrong: we're passing a JavaScript number but matching against a BN. We should have a different error message for these, the current one reads:

```
AssertionError: expected event argument 'value' to have value 42 but got 42: expected '42' to equal 42
```

Reply:
That is right, but the "expected '42' to equal 42" part is what Chai gives you when the types don't match anyways. If you believe we should have a manual check of types before passing the values to the chai assertion, let me know. I think this change belongs to a separate pull request, though.

Reply:
Agreed, and it's not like we didn't have this issue before these changes anyway. Thanks!